js/widgets.js

```diff
@@ -203,52 +229,51 @@ function embedDatasets (territories, datasets, dataTerritoryIdAttr, dataDatasetI
 }
 
 /**
- * Display matching datasets when the input search is filled.
+ * Display only matching datasets given the value of the search input.
  */
-function handleKeyup (territoryElement, searchNode, datasets, initialDatasets) {
-  searchNode.addEventListener('keyup', (searchEvent) => {
-    const searchValue = searchEvent.target.value
-    if (searchValue.length >= 3) {
-      const scoredResults = datasets.map((dataset) => {
-        const title = removeDiacritics(dataset.querySelector('.udata-title').innerText)
-        return {
-          dataset: dataset,
-          score: title.score(removeDiacritics(searchValue))
-        }
-      })
-      scoredResults.sort((a, b) => a.score > 0 && a.score < b.score)
-      const filteredResults = scoredResults.filter((result) => result.score > 0)
-      if (filteredResults) {
-        territoryElement.innerHTML = ''
-        filteredResults.forEach((result) => {
-          territoryElement.appendChild(result.dataset)
-        })
-      } else {
-        territoryElement.innerHTML = initialDatasets
+function filterDatasets (territoryElement, event, datasets, initialDatasets) {
+  const searchValue = event.target.value
+  if (searchValue.length >= 3) {
+    const scoredResults = datasets.map((dataset) => {
+      const title = removeDiacritics(dataset.querySelector('.udata-title').innerText)
+      return {
+        dataset: dataset,
+        score: title.score(removeDiacritics(searchValue))
       }
+    })
+    scoredResults.sort((a, b) => a.score > 0 && a.score < b.score)
+    const filteredResults = scoredResults.filter((result) => result.score > 0)
```

Comment:
You should probably filter before sorting, that would save some unneeded sort operations.